src/app/submission/sections/license/section-license.component.ts

```diff
@@ -0,0 +1,156 @@
+  styleUrls: ['./section-license.component.scss'],
+  templateUrl: './section-license.component.html',
+})
+@renderSectionFor(SectionsType.License)
+export class LicenseSectionComponent extends SectionModelComponent {
+
+  public formId;
+  public formModel: DynamicFormControlModel[];
+  public formLayout: DynamicFormLayout = SECTION_LICENSE_FORM_LAYOUT;
+  public displaySubmit = false;
+  public licenseText: string;
+
+  protected pathCombiner: JsonPatchOperationPathCombiner;
+  protected subs: Subscription[] = [];
+
+  @ViewChild('formRef') private formRef: FormComponent;
+
+  constructor(protected changeDetectorRef: ChangeDetectorRef,
+              protected collectionDataService: CollectionDataService,
+              protected formBuilderService: FormBuilderService,
+              protected formOperationsService: SectionFormOperationsService,
+              protected formService: FormService,
+              protected operationsBuilder: JsonPatchOperationsBuilder,
+              protected sectionService: SectionsService,
+              protected submissionService: SubmissionService,
+              @Inject('collectionIdProvider') public injectedCollectionId: string,
+              @Inject('sectionDataProvider') public injectedSectionData: SectionDataObject,
+              @Inject('submissionIdProvider') public injectedSubmissionId: string) {
+    super(injectedCollectionId, injectedSectionData, injectedSubmissionId);
+  }
+
+  onSectionInit() {
+    this.pathCombiner = new JsonPatchOperationPathCombiner('sections', this.sectionData.id);
+    this.formId = this.formService.getUniqueId(this.sectionData.id);
+    this.formModel = this.formBuilderService.fromJSON(SECTION_LICENSE_FORM_MODEL);
+    const model = this.formBuilderService.findById('granted', this.formModel);
+
+    this.subs.push(
+      this.collectionDataService.findById(this.collectionId).pipe(
+        filter((collectionData: RemoteData<Collection>) => isNotUndefined((collectionData.payload))),
+        flatMap((collectionData: RemoteData<Collection>) => collectionData.payload.license),
+        find((licenseData: RemoteData<License>) => isNotUndefined((licenseData.payload))))
+        .subscribe((licenseData: RemoteData<License>) => {
+          this.licenseText = licenseData.payload.text;
+
+          // Retrieve license accepted status
+          if ((this.sectionData.data as WorkspaceitemSectionLicenseObject).granted) {
+            (model as DynamicCheckboxModel).valueUpdates.next(true);
+          } else {
+            (model as DynamicCheckboxModel).valueUpdates.next(false);
+          }
+
+          // Disable checkbox whether it's in workflow or item scope
+          this.sectionService.isSectionReadOnly(
+            this.submissionId,
+            this.sectionData.id,
+            this.submissionService.getSubmissionScope()
+          ).pipe(
+            take(1),
+            filter((isReadOnly) => isReadOnly))
+            .subscribe(() => {
```

Comment:
This subscribe is never unsubscribed. It is also nested inside another subscribe, so it can probably be rewritten using switchMap

Reply:
fixed